recipes/mingw/all/conanfile.py

```diff
@@ -0,0 +1,44 @@
+import os
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+
+
+class MingwConan(ConanFile):
+    name = "mingw"
+    description = "MinGW is a contraction of Minimalist GNU for Windows"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://www.mingw.org"
+    license = "ZPL-2.1", "MIT", "GPL-2.0-or-later"
+    topics = ("gcc", "gnu", "unix", "mingw32", "binutils")
+    settings = "os", "arch"
+    options = {"threads": ["posix", "win32"], "exception": ["seh", "sjlj"]}
```

Comment:
This means we allow option mingw:threads and mingw:exception to be different from settings compiler.threads and compiler.exception. This sounds very dangerous for the user

Reply:
Yes, but unfortunately this is the only way right now. We are using the same approach for Android NDK. With new profiles approach (build x target) we still can't solve it, but it's listed to be solve on Conan 2.0.

The recommendation will be generate a profile which combines both settings and options.

Reply:
Which are NDK are you referring to ? https://github.com/conan-io/conan-center-index/blob/master/recipes/android-ndk/all/conanfile.py does not seem to use options.
At the very least we should print a warning if the self.settings.compiler is not "gcc", and if one of the options does not match the settings

Reply:
Yes, a warning will be made

Reply:
We can add a check in the package_info when using two-profiles approach:

```python

def package_info(self):
    if hasattr(self, 'settings_target'):
        assert self.settings_target.compiler == 'gcc'
        if self.settings_target.compiler.threads != self.options.threads:
            raise ConanInvalidConfiguration('Build requires 'mingw' provides binaries for gcc with threads={}, your profile:host declares threads={}, please....')
        ....

```

I don't know how friendly the exception will be, but at least the message should tell the user information about the mismatch in the settings and options.

Reply:
That's what I was looking for!